sentinel/src/schedule/outbound.js

```diff
@@ -36,15 +38,37 @@ const queuedTasks = () =>
       keys: associatedDocIds,
       include_docs: true
     }).then(results => {
-      const associatedDocs = results.rows.map(r => r.doc);
-      return lineage.hydrateDocs(associatedDocs);
-    }).then(associatedDocs => {
-      // allDocs returns results in order, as does hydrateDocs, so we can just
-      // combine them with the task docs above
-      return outboundTaskDocs.map((t, idx) => ({
-        taskDoc: t,
-        medicDoc: associatedDocs[idx]
-      }));
+      const { validTasks, invalidTasks } = results.rows.reduce((acc, r, idx) => {
+        const task = outboundTaskDocs[idx];
+        if (r.doc) {
+          acc.validTasks.push({
+            task: task,
+            doc: r.doc
+          });
+        } else if (r.error === 'not_found') {
```

Comment:
I sometimes get responses that look like this when requesting deleted documents:
```
{ id: 'random_message_26a0e8ff-b3ff-45d7-9226-003f32deffe0',
  key: 'random_message_26a0e8ff-b3ff-45d7-9226-003f32deffe0',
  value: { rev: '4-3b2984a729f644a426980e1a08bc310c', deleted: true },
  doc: null }
```
and the code ends up throwing errors and not removing the tasks.